modules/gapi/src/backends/fluid/gfluidcore.cpp

```diff
@@ -797,6 +797,51 @@ GAPI_FLUID_KERNEL(GFluidDivRC, cv::gapi::core::GDivRC, false)
     }
 };
 
+//-------------------
+//
+// Fluid kernels: mask
+//
+//-------------------
+
+template<typename DST, typename SRC>
+static void run_mask(Buffer &dst, const View &src, const View &mask)
+{
+    static_assert(std::is_same<DST, SRC>::value, "wrong types");
+
+    int width  = dst.length();
+    int chan   = dst.meta().chan;
+    int length = width * chan;
+
+    if (mask.meta().chan != 1 || mask.meta().depth != CV_8U)
+        CV_Error(cv::Error::StsBadArg, "unsupported mask type");
+
+    const auto *in1 = src.InLine<SRC>(0);
+    const auto *in2 = mask.InLine<uchar>(0);
```

Comment:
I would rename `in2` to something else to match its purpose (e.g. `in_mask`). It's a little confusing below: `out[l] = in2[l] ? in1[l] : 0`

Reply:
+1

Reply:
Agree, changed. Thanks!